editor/editor_node.cpp

```diff
@@ -1605,16 +1605,56 @@ static bool _find_edited_resources(const Ref<Resource> &p_resource, Set<Ref<Reso
 	p_resource->get_property_list(&plist);
 
 	for (const PropertyInfo &E : plist) {
-		if (E.type == Variant::OBJECT && E.usage & PROPERTY_USAGE_STORAGE && !(E.usage & PROPERTY_USAGE_RESOURCE_NOT_PERSISTENT)) {
-			RES res = p_resource->get(E.name);
-			if (res.is_null()) {
-				continue;
-			}
-			if (res->get_path().is_resource_file()) { // not a subresource, continue
-				continue;
-			}
-			if (_find_edited_resources(res, edited_resources)) {
-				return true;
+		switch (E.type) {
+			case Variant::ARRAY: {
+				Array varray = p_resource->get(E.name);
+				int len = varray.size();
+				for (int i = 0; i < len; i++) {
+					const Variant &v = varray.get(i);
+
+					if (v.get_type() == Variant::OBJECT) {
+						RES res = v;
```

Comment:
Is `v` guaranteed to be a valid `Ref<Resource>` in here? If not then it needs to be checked (because `_find_edited_resources` called below isn't checking for its argument validity):
```
if (res.is_null()) {
	continue;
}
```

Reply:
Good point. I tried a bit around but looks like an explicit Object inside a `Dictionary`/`Array` can't be null.
But since e.g. `ResourceFormatSaverTextInstance::_find_resources` is also doing a null check, it is probably the best to add it here as well.